code/code/misc/spell_parser.cc

```diff
@@ -2413,6 +2414,9 @@ int TBeing::doDiscipline(spellNumT which, sstring const& n1)
       case SPELL_SANCTUARY:
           sanctuary(this, ch);
         break;
+      case SPELL_CRUSADE:
```

Comment:
This monstrosity could be changed into a neat little lookup table of spellnum -> function.

Reply:
I'm trying to figure this out.  Like it seems like it'd be possible to make a map of spell number to a function but then calling the function that may have different return types is tripping me up. Any help?